addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -788,10 +856,201 @@ private void checkConnection() {
+                }
+            }
+            return null;
+        }
+
+        private void stopSerialReaders() {
+            if (deviceSupportsSerial1() && serialReader1 != null) {
+                serialReader1.stop();
+            }
+            if (deviceSupportsSerial2() && serialReader2 != null) {
+                serialReader2.stop();
+            }
+        }
+    }
+
+    /*
+     * The {@link SerialReader} class reads data from the serial connection. When data is
+     * received, the receive channel is updated with the data. Data is read up to the
+     * end-of-message delimiter defined in the Thing configuration.
+     *
+     * @author Mark Hilbush - Initial contribution
+     */
+    private class SerialReader {
+        private Logger logger = LoggerFactory.getLogger(SerialReader.class);
+
+        private CommandType serialPort;
+        private BufferedInputStream serialIn;
+        private ScheduledFuture<?> serialReaderJob;
+        private boolean terminateSerialReader;
+
+        private byte[] eom;
+
+        private Runnable serialReaderRunnable = new Runnable() {
+            @Override
+            public void run() {
+                serialReader();
+            }
+        };
+
+        SerialReader(CommandType serialPort, BufferedInputStream serialIn, byte[] eom) {
+            if (serialIn == null) {
+                throw new IllegalArgumentException("Serial input stream is not set");
+            }
+            this.serialPort = serialPort;
+            this.serialIn = serialIn;
+            this.eom = eom;
+            serialReaderJob = null;
+            terminateSerialReader = false;
+        }
+
+        public void start() {
+            serialReaderJob = scheduledExecutorService.schedule(serialReaderRunnable, 0, TimeUnit.SECONDS);
+        }
+
+        public void stop() {
+            if (serialReaderJob != null) {
+                terminateSerialReader = true;
+                serialReaderJob.cancel(true);
+                serialReaderJob = null;
+            }
+        }
+
+        private void serialReader() {
+            logger.info("Serial reader RUNNING for {} on {}:{}", thingID(), getIP(), serialPort);
+
+            while (!terminateSerialReader) {
+                byte[] buffer;
+                try {
+                    buffer = readUntilEndOfMessage(eom);
+                    if (buffer == null) {
+                        logger.debug("Received end-of-stream from {} on {}", getIP(), serialPort);
+                        continue;
+                    }
+                    logger.debug("Rcv data from {} at {}:{}: {}", thingID(), getIP(), serialPort,
+                            getAsHexString(buffer));
+                    updateFeedbackChannel(buffer);
+
+                    Thread.sleep(100);
+
+                } catch (UnsupportedEncodingException e) {
+                    logger.info("Unsupported encoding exception: {}", e.getMessage(), e);
+                    continue;
```

Comment:
I debated this.  I came down on the side of being explicit, so as to not make one wonder if I overlooked that catch clause compared to the other 2 below it.  I can go either way on it, so I'll defer to your best judgement.